pkgs/nix-tools/darwin-rebuild.sh

```diff
@@ -92,22 +112,48 @@ done
 
 if [ -z "$action" ]; then showSyntax; fi
 
-if ! [ "$action" = build ]; then
+flakeFlags=(--experimental-features 'nix-command flakes')
+
+if [ -n "$flake" ]; then
+    if [[ $flake =~ ^(.*)\#([^\#\"]*)$ ]]; then
+       flake="${BASH_REMATCH[1]}"
+       flakeAttr="${BASH_REMATCH[2]}"
+    fi
+    if [ -z "$flakeAttr" ]; then
+      flakeAttr=$(hostname)
```

Comment:
Will this break if I add the hostname attribute?

Reply:
Not exactly sure what the question is, but this is the fallback in case the flake-ref doesn't include a hash. So these are equivalent.

```sh
darwin-rebuild switch --flake ~/.config/darwin
darwin-rebuild switch --flake ~/.config/darwin#Johns-MacBook.local
```

I did just notice that  on NixOS the nixosConfigurations prefix is always added, which makes sense so I'll fix that.

Reply:
That was my question, did not see the fallback switch.

(Also not a fan of always prefixing it, but that's probably something to discuss at nixpkgs)

Reply:
I think it makes sense to namespace sets by their type but that makes accessing them a bit cumbersome.